### PR TITLE
Remove redundant checks and reduce dependencies for generating swap messages

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs [hunk 1]
```diff
@@ -58,7 +58,11 @@ pub fn refund_bank_msg(
 
 /// swap will always swap over the CL pool. In the future we may expand the
 /// feature such that it chooses best swaps over all routes
-pub fn swap_msg(deps: &DepsMut, env: &Env, params: SwapParams) -> Result<CosmosMsg, ContractError> {
+pub fn swap_msg(
+    deps: &DepsMut,
+    sender: String,
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs:63]: Changing the type to `Addr` improves type safety

File: smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs [hunk 2]
```diff
@@ -91,14 +95,14 @@ pub fn swap_msg(deps: &DepsMut, env: &Env, params: SwapParams) -> Result<CosmosM
 }
 
 fn osmosis_swap_exact_amount_in_msg(
-    env: &Env,
+    sender: String,
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs:98]: Changing the type to `Addr` improves type safety

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs [hunk 3]
```diff
@@ -34,7 +34,7 @@ pub struct SwapParams {
 
 pub fn execute_swap_non_vault_funds(
     deps: DepsMut,
-    env: Env,
+    contract_address: String,
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs:37]: Changing the type to `Addr` improves type safety

**Review by LaurensKubat — CHANGES_REQUESTED** [review]
in general lgtm, 1 small comment on accepting addresses as `Addr` instead of string

File: smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs]: done

File: smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs (hunk 2 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/helpers/msgs.rs]: done

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs (hunk 3 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs]: done

**Review by LaurensKubat — DISMISSED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request merged**